presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java

```diff
@@ -208,22 +208,22 @@ public void getSetBits(Type type, int batchSize, BlockBuilder builder)
     public void getSetBits(Type type, int batchSize, BlockBuilder builder, boolean[] isNull)
             throws IOException
     {
-        for (int i = 0; i < batchSize; i++) {
+        getSetBits(type, batchSize, builder, isNull, 0);
+    }
+
+    /**
+     * Sets the vector element to true for the batchSize number of elements starting at offset
+     * if the bit is set, skipping the null values.
+     */
+    public void getSetBits(Type type, int batchSize, BlockBuilder builder, boolean[] isNull, int offset)
+            throws IOException
+    {
+        for (int i = offset; i < batchSize + offset; i++) {
             if (isNull[i]) {
                 builder.appendNull();
             }
             else {
-                // read more data if necessary
-                if (bitsInData == 0) {
-                    readByte();
-                }
-
-                // read bit
-                type.writeBoolean(builder, (data & HIGH_BIT_MASK) != 0);
-
-                // mark bit consumed
-                data <<= 1;
-                bitsInData--;
+                type.writeBoolean(builder, nextBit());
```

Comment:
I will consider making this refactor a separate refactor commit.  This in general helps review.

Reply:
I will have a seperate commit to do more refactoring